src/main/webapp/app/complaints/complaint.service.ts

```diff
@@ -260,7 +260,12 @@ export class ComplaintService implements IComplaintService {
         }
         const complaintStartDate = dayjs.max(possibleComplaintStartDates);
 
-        return dayjs().isBefore(complaintStartDate) ? undefined : complaintStartDate.add(complaintTimeFrame, 'days');
+        if (dayjs().isBefore(complaintStartDate)) {
+            return undefined;
+        }
+        if (complaintStartDate != undefined) {
```

Comment:
```suggestion
        if (complaintStartDate) {
```

Reply:
I'm a bit confused about this place in general. If `complaintStartDate` actually is undefined, the call above would not make much sense (`dayjs().isBefore(undefined)` always evaluates to false). So we don't return `undefined` above, but also don't return the value inside this case. In the end we implicitly return undefined, but without calling any return statement.

Reply:
feel free to propose an alternative. The method can return undefined and I would not worry too much about 

`if (complaintStartDate != undefined)` vs `if (complaintStartDate)`

the dayjs API changed and max can now also return undefined so this change was necessary